torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -2,9 +2,15 @@
 
 #include <gloo/allreduce_halving_doubling.h>
 #include <gloo/broadcast_one_to_all.h>
+#include <gloo/cuda_allreduce_halving_doubling.h>
+#include <gloo/cuda_broadcast_one_to_all.h>
 #include <gloo/rendezvous/context.h>
 #include <gloo/transport/tcp/device.h>
 
+#include <THC.h>
+
+#include "private/CUDAUtils.hpp"
```

Comment:
This means gloo will only be possible to compile when someone has CUDA. Do we really want to enforce that?

Reply:
For now, I'm fine with this.

I want to get a first rev out and making it depend on CUDA is reasonable IMO.